addons/base_automation/models/base_automation.py

```diff
@@ -80,6 +83,13 @@ class BaseAutomation(models.Model):
         readonly=False,
     )
     active = fields.Boolean(default=True, help="When unchecked, the rule is hidden and will not be executed.")
+
+    @api.constrains("trigger", "model_id")
+    def _check_trigger(self):
+        for automation in self:
+            if automation.trigger in MAIL_TRIGGERS and not automation.model_id.is_mail_thread:
+                raise exceptions.ValidationError(_("Reacting on mail events are only available on model implementing Mail Thread."))
```

Comment:
There is little need to be super comprehensive for 'lambda' users, since the faulty configuration leading to this message cannot be achieved via the UI (the mail options in the selection field for 'trigger' are not shown if the model does not inherit mail.thread)

Though including the name of the model is a good idea in general, certainly - my point is not to spend too much time on this, if someone gets this error it's because they screwed around in technical shit